snapcraft/internal/repo/_deb.py

```diff
@@ -186,8 +188,100 @@ def get_packages_for_source_type(cls, source_type):
+                                       '644', sources_lists])
+
+        try:
+            update_output = subprocess.check_output(
+                ['sudo', 'apt-get', 'update'],
+                stderr=subprocess.STDOUT).decode(sys.getfilesystemencoding())
+            # Failure to download doesn't return an error code
+            if 'Err:' in update_output:
+                raise subprocess.CalledProcessError(
+                    100, ['sudo', 'apt-get', 'update'], update_output)
+        except subprocess.CalledProcessError as e:
+            logger.error(e.output)
+            raise e
+        return package_name in apt_cache
+
+    @classmethod
+    def _get_build_deps(cls, package_names, arch):
+        """Use apt-get build-dep with a fake source package file to find
+           all dependencies and correct architectures.
+        """
+        if not package_names:
+            return []
+        with tempfile.NamedTemporaryFile(suffix='.dsc') as fake_source:
+            depends = 'Build-Depends: {}\n'.format(', '.join(package_names))
+            fake_source.write(depends.encode())
+            fake_source.flush()
+            build_deps = []
+            try:
+                actions = subprocess.check_output(
+                    ['apt-get', 'build-dep', '-q', '-s',
+                     '-a{}'.format(arch), fake_source.name],
+                    stderr=subprocess.STDOUT, env={}).decode(
+                        sys.getfilesystemencoding())
+                msg = 'The following NEW packages will be installed:\n  '
+                msg_end = len(msg) + actions.find(msg)
+                packages_end = re.search('\d+ upgraded', actions).start()
+                build_deps = actions[msg_end:packages_end].split()
+            except subprocess.CalledProcessError as e:
+                actions = e.output.decode(sys.getfilesystemencoding())
+                rx = re.compile(
+                    '(.+Depends: (.+) but it is not .+|.+)')
+                for line in actions.split('\n'):
+                    build_deps.append(rx.sub('\\2', line))
+            # apt-get build-dep will output name=version:arch
+            # Ensure that the format is always name:arch=version
+            pkgs = []
+            for pkg in build_deps:
+                if not pkg:
+                    continue
+                name, version_with_arch = repo.get_pkg_name_parts(pkg)
+                if version_with_arch:
+                    if ':' in version_with_arch:
+                        version, arch = version_with_arch.split(':')
+                        name += ':{}'.format(arch)
+                    else:
+                        version = version_with_arch
+                    name += '={}'.format(version)
+                pkgs.append(name)
+            return pkgs
+
+    @classmethod
+    def install_build_packages(cls, package_names, arch):
+        unique_packages = set(cls._get_build_deps(package_names, arch))
```

Comment:
This is crashing for me:

 > /home/ubuntu/workspace/canonical/snapcraft/snapcraft/internal/repo/_deb.py(286)install_build_packages()
-> new_packages = []
(Pdb) unique_packages
{'for', 'Keep', 'privileges', 'so', 'package', 'get', 'build', 'in', 'relevance', "don't", "'/tmp/tmpvnu0mvxc.dsc'", 'tree...', 'file', 'the', 'is', 'dependencies', 'depend', 'real', 'on', 'also', 'deactivated,', 'locking', 'execution.', 'Reading', 'information...', 'current', 'needs', 'Building', 'mind', 'using', 'root', 'state', 'to', 'situation!', 'lists...', 'that', 'Note,', 'dependency'}

$ apt-get build-dep -q -s -aarm64 golang-go:native        
NOTE: This is only a simulation!
      apt-get needs root privileges for real execution.
      Keep also in mind that locking is deactivated,
      so don't depend on the relevance to the real current situation!
E: No architecture information available for arm64. See apt.conf(5) APT::Architectures for setup